### ISSUE TITLE
Proposal/Enhancement: Upgrading to Newer YOLO Architecture & Edge Optimization

### ISSUE THREAD

**Issue by KilianMa** [varungupta31/dashcam_anonymizer#14]
Hi Varun,

Greetings from Germany, and thanks for providing this repository. We arre currently using your anonymization pipeline for our Sensorbike project at university.

In your README, you mentioned being open to improvements. I am planning to train a newer, optimized YOLO architecture to improve inference times and detection accuracy.

To achieve the best possible baseline and ensure backward compatibility with your pipeline, I would ideally like to train the new model on the exact same dataset you used for best.pt, or fine-tune it accordingly.

Would you be willing to share the dataset, or point me towards the sources you used for training? In that case, any details on your training data composition or hyperparameters would be highly appreciated!

I would be happy to contribute my optimized model weights back via a Pull Request once they are ready.

Best regards,
Kilian

**Comment by varungupta31** [comment]
Hey @KilianMa !

Thanks for your attention to this work :)

Dataset, nope - can't find it anymore, sorry.
Uhm, it was mostly compiled from the License Plates and Human/Face/Person category from [OpenImagesDataset](https://storage.googleapis.com/openimages/web/index.html) in pre-claude times, and came with bbox annotations which I converted to YOLO format and trained.

However, not to discard my own work, but you may want to take a look at
https://github.com/facebookresearch/EgoBlur

This perhaps would be more optimized as is, and is used a bit more widely - and can serve your purpose without additional work.


Happy to merge possible improvements if you still proceed!

Good Luck,
Varun
